### ISSUE TITLE
Initial rendering of emoji in Text is very slow

### ISSUE THREAD

**Issue by yunyu** [flutter/flutter#27328]
I have a ListView consisting of several Text children containing large amounts of emojis, and the initial render for any given string passed into the Text is very slow. 

I am on Flutter version 1.1.8, which includes a commit that speeds up subsequent renders (https://github.com/flutter/engine/pull/7208), but there is still extremely noticeable stuttering (lasting up to 1/4th of a second) when an emoji-heavy string is first rendered.

Would it be possible to do something like pre-populate the font cache with all emojis (e.g. https://stackoverflow.com/a/36258684/1807358), or make the cache per-code-point-range rather than per-character?

**Comment by zoechi** [comment]
Please add the output of `flutter doctor -v`.

**Comment by yunyu** [comment]
Here is the output:

```
[√] Flutter (Channel beta, v1.1.8, on Microsoft Windows [Version 10.0.17763.292], locale en-US)
    • Flutter version 1.1.8 at C:\Users\Yunyu\flutter
    • Framework revision 985ccb6d14 (3 weeks ago), 2019-01-08 13:45:55 -0800
    • Engine revision 7112b72cc2
    • Dart version 2.1.1 (build 2.1.1-dev.0.1 ec86471ccc)

[√] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at C:\Users\Yunyu\AppData\Local\Android\sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)
    • All Android licenses accepted.

[!] Android Studio (version 3.2)
    • Android Studio at C:\Program Files\Android\Android Studio
    X Flutter plugin not installed; this adds Flutter specific functionality.
    X Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)

[!] IntelliJ IDEA Ultimate Edition (version 2018.1)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA 2018.1
    X Flutter plugin not installed; this adds Flutter specific functionality.
    X Dart plugin not installed; this adds Dart specific functionality.
    • For information about installing plugins, see
      https://flutter.io/intellij-setup/#installing-the-plugins

[√] VS Code, 64-bit edition (version 1.30.2)
    • VS Code at C:\Program Files\Microsoft VS Code
    • Flutter extension version 2.22.2

[√] Connected device (1 available)
    • ONEPLUS A3000 • 67b32577 • android-arm64 • Android 8.0.0 (API 26)

! Doctor found issues in 2 categories.
```

**Comment by jason-simmons** [comment]
Most of the time during scrolling is spent in the Minikin text layout engine.

I did some profiling and noticed that a major bottleneck is the `isColorBitmapFont` function in `Layout.cpp`.  It's calling `hb_face_reference_table` to check for a color bitmap data table, and it looks like HarfBuzz is copying the table even though we only need to check for its presence.  For the NotoColorEmoji font, this table is apparently several megabytes.

I don't know if HarfBuzz has a faster way to check if a font is a color bitmap font.  I tried calling `hb_face_get_table_tags` but it didn't return results for the fonts I tested.

If there isn't a faster way, then we could cache the results of `isColorBitmapFont`
